crates/agent/src/thread.rs

```diff
@@ -1483,24 +1486,40 @@ impl Thread {
         canceled
     }
 
-    /// Returns the feedback given to the thread, if any.
+    /// Returns the overall feedback given to the thread, if any.
     pub fn feedback(&self) -> Option<ThreadFeedback> {
         self.feedback
     }
 
-    /// Reports feedback about the thread and stores it in our telemetry backend.
-    pub fn report_feedback(
+    /// Returns the feedback for a specific message, if any.
+    pub fn message_feedback(&self, message_id: MessageId) -> Option<ThreadFeedback> {
+        self.message_feedback.get(&message_id).copied()
+    }
+
+    /// Reports feedback about a specific message and stores it in our telemetry backend.
+    /// Does NOT affect the overall thread feedback.
+    pub fn report_message_feedback(
         &mut self,
+        message_id: MessageId,
         feedback: ThreadFeedback,
         cx: &mut Context<Self>,
     ) -> Task<Result<()>> {
         let final_project_snapshot = Self::project_snapshot(self.project.clone(), cx);
         let serialized_thread = self.serialize(cx);
         let thread_id = self.id().clone();
         let client = self.project.read(cx).client();
-        self.feedback = Some(feedback);
+
+        // Store feedback for this specific message only
+        self.message_feedback.insert(message_id, feedback);
+
         cx.notify();
 
+        // Get the message content
```

Comment:
Can we remove these comments? They're restating what the code does, which adds a lot of extra noise.